### DISCUSSION TITLE
Terminology: collision_margin and collision_margin_buffer

### DISCUSSION THREAD

**Discussion by rjoomen** [tesseract-robotics/trajopt#347] [Q&A] [answered]
I was reading the original Trajopt paper, and for collision checking they talk about `d_safe` and `d_check`.

My understanding is that in this implementation, `d_safe ≡ collision_margin` and `d_check ≡ collision_margin + collision_margin_buffer`, is that correct?

**Comment by Levi-Armstrong** [comment, marked as answer]
It is interesting that the paper mentions it but the original implementation did not have it because I added the buffer due to stability issues, but to answer your question your understanding is correct. This is so you still have an equation in the optimization when just outside the contact margin.

**Comment by rjoomen** [comment]
Strange. Or was maybe [this](https://github.com/tesseract-robotics/trajopt/pull/153) the old implementation of a buffer?

**Reply by Levi-Armstrong** [reply]
Yea, that may have been it.
